pkg/apis/clusteroperator/validation/clusterdeployment.go

```diff
@@ -86,6 +86,13 @@ func validateClusterDeploymentSpec(spec *clusteroperator.ClusterDeploymentSpec,
 		allErrs = append(allErrs, field.Required(versionPath.Child("name"), "must specify a cluster version to install"))
 	}
 
+	if len(spec.NetworkConfig.Services.CIDRBlocks) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkConfig").Child("services"), &spec.NetworkConfig.Services, "only one service network subnet is currently supported"))
+	}
+	if len(spec.NetworkConfig.Pods.CIDRBlocks) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkConfig").Child("pods"), &spec.NetworkConfig.Pods, "only one pod network subnet is currently supported"))
+	}
+
```

Comment:
We should also do the same validation that's done upstream:
https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/apis/cluster/v1alpha1/cluster_types.go#L118-L136

Reply:
I've got defaults being added for all three of those fields, do we need validation in that case or can I assume those defaults will *always* be applied?